fix input autofill colors

alexey-petrov opened the pull request:
Closes:
- https://github.com/sparkletown/internal-sparkle-issues/issues/1579

Reproduced using Win10/Mac Chrome

Styles differ a bit in multiple components we have. Tried to match the background color based on other inputs in corresponding forms ( admin settings, timing events, user search etc )

Also, added a guard to event timing list ( used to break the page if event missed some details )

Before:
https://www.loom.com/share/6d213a11642247c29bc4e3e098e04be2

After: 
![image](https://user-images.githubusercontent.com/56254806/145375374-ef067e41-5be2-402e-9e98-9a8722d7df1b.png)

alexey-petrov commented on src/utils/event.ts:
```diff
@@ -60,11 +60,21 @@ export const eventHappeningNow = (
-export const eventStartTime = (event: VenueEvent) =>
-  fromUnixTime(event.start_utc_seconds);
+export const eventStartTime = (event: VenueEvent) => {
+  if (!event?.start_utc_seconds) {
```
had some events with missing details, so had to add this guard to prevent timing page from breaking

goran-peoski-work replied:
Will this impact other calls of the same utility function?

goran-peoski-work replied:
Consider this: pure function got turned into impure one.
Now none of  the places it is called from can depend on it to consistently return the same result for the same input.

A way to make it pure again is to not generate input inside, but add second options parameter with field `defaultValue` for the places that want this kind of behavior

colinhowe replied:
Or, just fix your events ;)

alexey-petrov commented on src/utils/event.ts:
```diff
@@ -60,11 +60,21 @@ export const eventHappeningNow = (
 
-export const eventEndTime = (event: VenueEvent) =>
-  addMinutes(eventStartTime(event), event.duration_minutes);
+export const eventEndTime = (event: VenueEvent) => {
```
had some events with missing details, so had to add this guard to prevent timing page from breaking

goran-peoski-work commented on src/components/atoms/InputField/InputField.scss:
```diff
@@ -33,6 +33,17 @@ $icon-width: 2.5rem;
+    &:-webkit-autofill:focus,
+    &:-webkit-autofill:active {
+      -webkit-text-fill-color: var(--content--over);
+      -webkit-box-shadow: 0 0 0 $input-auto-fill-box-shadow-height
+        $input-auto-fill-box-shadow inset;
+      box-shadow: 0 0 0 $input-auto-fill-box-shadow-height
+        $input-auto-fill-box-shadow inset;
```
Please make a single `box-shadow--autofilled` similar to https://github.com/sparkletown/sparkle/blob/staging/src/scss/constants.scss#L313 for reuse here and in the other components

alexey-petrov replied:
done 917110d

goran-peoski-work commented on src/components/organisms/TimingDeleteModal/TimingDeleteModal.scss:
```diff
@@ -0,0 +1,14 @@
+@import "scss/constants.scss";
+
+.form {
```
This should probably be encapsulated inside a component class, like
```suggestion
TimingDeleteModal {
  .form {
```

alexey-petrov replied:
done 917110d

goran-peoski-work commented on src/components/organisms/TimingDeleteModal/TimingDeleteModal.tsx:
```diff
@@ -75,25 +77,29 @@ export const TimingDeleteModal: React.FC<TimingDeleteModalProps> = ({
 
   return (
     <Modal show={show} onHide={onHide}>
-      <div className="form-container">
+      <div className="form">
```
For consistency with other components.

One way would be:

```suggestion
      <div className="TimingDeleteModal TimingDeleteModal__form">
```

although that leaves the top level component not the same class as this component, so

another way would be:

```suggestion
      <div className="TimingDeleteModal">
        <Modal show={show} onHide={onHide}>
          <div className="TimingDeleteModal__form">
```

alexey-petrov replied:
done 917110d

goran-peoski-work commented on src/scss/constants.scss:
```diff
@@ -33,6 +33,12 @@ $lighter-intermediate-grey: #3d3a3f;
 $light-grey: #292929;
 $light-grey--lightest: #acadb0;
 
+// @debt probably needs to be replaced with var(...), but it's used in a place
+// where var wasn't working
+$input-auto-fill-box-shadow: $secondary--dark;
+$input-auto-fill-box-shadow-timeline: #434343;
+$input-auto-fill-box-shadow-height: 30px;
+
```
Delete. Not only replaced with var, but also with a `box-shadow--autofilled` function. Please don't use `$secondary-dark` there should be a comment after that color for suitable replacement. Similar to the gray `#434343` one, there are already existing ones at `:root`

alexey-petrov replied:
removed and replaced with var() 9bff7d8

goran-peoski-work reviewed (changes requested):
Few suggestions and comments. Let's do a little screen share about the colors when you can.

alexey-petrov commented:
@goran-peoski-work as for the issue described in https://github.com/sparkletown/sparkle/pull/2695#discussion_r765700244

I've decided against those changes and reverted the recently added guards. Since newer events that we create are validated to have all the necessary `date` attributes that page won't be crashing, but I've added @debt comments there, so that we can add some sort of `default` value or guard this piece of code from crashing in some other way

goran-peoski-work commented on src/components/molecules/AdminInput/AdminInput.scss:
```diff
@@ -24,10 +24,9 @@
-        $input-auto-fill-box-shadow inset;
-      box-shadow: 0 0 0 $input-auto-fill-box-shadow-height
-        $input-auto-fill-box-shadow inset;
+      -webkit-box-shadow: box-shadow--input-shadow(var(--content--under-80a))
+        inset;
+      box-shadow: box-shadow--input-shadow(var(--content--under-80a)) inset;
```
Kind of forgot these would still be duplicates. Don't know why override it with the prefix though, may want to reconsider  at least adding a `@debt` comment to recheck `caniuse.com` in the future and deal with it. Browsers should eventually just use the standard one.

alexey-petrov replied:
removed `--webkit` ones. confirmed that everything is working fine c974b57

goran-peoski-work commented on src/scss/constants.scss:
```diff
@@ -320,6 +318,10 @@ $z-layers: (
   @return 0 0 0 2px $color;
 }
 
+@function box-shadow--input-shadow($color) {
```
Not the luckiest pick of name I think, sure it is connected to input, but shadow appearing twice in the name...

alexey-petrov replied:
renamed c974b57

goran-peoski-work commented on src/scss/constants.scss:
```diff
@@ -35,9 +35,7 @@ $light-grey--lightest: #acadb0;
 // @debt probably needs to be replaced with var(...), but it's used in a place
 // where var wasn't working
-$input-auto-fill-box-shadow: $secondary--dark;
 $input-auto-fill-box-shadow-timeline: #434343;
```
This one should be removed.

```suggestion
```

alexey-petrov replied:
removed

goran-peoski-work commented on src/scss/constants.scss:
```diff
@@ -35,9 +35,7 @@ $light-grey--lightest: #acadb0;
 
 // @debt probably needs to be replaced with var(...), but it's used in a place
 // where var wasn't working
```
```suggestion
```

goran-peoski-work reviewed (approved):
Approving, with: 

- you can add `@debt` comments if this PR is needed fast and we can rework some of the logic later
- make pretty if something can be polished in 5-10 minutes, I guess, then merge

alexey-petrov commented on src/utils/event.ts:
```diff
@@ -60,21 +57,15 @@ export const eventHappeningNow = (
+// @debt: if event or event.start_utc_seconds are falsy this func crashes the page
+// probably need to add an additional `defaultValue` to return or some sort of a guard
+export const eventEndTime = (event: VenueEvent) =>
+  addMinutes(eventStartTime(event), event.duration_minutes);
```
changed props + types c974b57
and updated these functions usages everywhere in the codebase
